### PR TITLE
docs: split reduce and fold in two different files

### PR DESCRIPTION
**Pull request by etiennebacher**

### CODE REVIEW

File: R/functions__lazy.R [hunk 1]
```diff
@@ -841,11 +841,9 @@ pl_rolling_corr = function(a, b, window_size, min_periods = NULL, ddof = 1) {
 
 #' Accumulate over multiple columns horizontally with an R function
 #'
-#' @description `pl$fold()` and `pl$reduce()` allows one to do rowwise operations.
-#' The only difference between them is that `pl$fold()` has an additional argument
-#' (`acc`) that contains the value that will be initialized when the fold starts.
-#'
-#' @name pl_fold_reduce
+#' This allows one to do rowwise operations, starting with an initial value
+#' (`acc`). See `pl$reduce()` to do rowwise operations without this initial
```
Comment (eitsupi) [inline R/functions__lazy.R:845]:
```suggestion
#' (`acc`). See [`pl$reduce()`][pl_reduce] to do rowwise operations without this initial
```

File: R/functions__lazy.R [hunk 2]
```diff
@@ -856,26 +854,35 @@ pl_rolling_corr = function(a, b, window_size, min_periods = NULL, ddof = 1) {
...
   fold(acc, lambda, exprs) |>
     unwrap("in pl$fold():")
 }
 
-#' @rdname pl_fold_reduce
-#' @name pl_fold_reduce_part2
+#' @inherit pl_fold title params return
+#'
+#' @description
+#' This allows one to do rowwise operations. See `pl$fold()` to do rowwise
```
Comment (eitsupi) [inline R/functions__lazy.R:873]:
```suggestion
#' This allows one to do rowwise operations. See [`pl$fold()`][pl_fold] to do rowwise
```

**Review by eitsupi — CHANGES_REQUESTED** [review]
Thanks!
Could you fix cross refs?

**Review by eitsupi — APPROVED** [review]

**Comment by etiennebacher** [comment]
Thanks for the review

**Pull request merged**